salt/modules/ldap3.py

```diff
@@ -484,6 +503,12 @@ def modify(connect_spec, dn, directives):
     # not modified)
     modlist = [(getattr(ldap, 'MOD_' + op.upper()), attr, list(vals))
                for op, attr, vals in directives]
+
+    for idx, mod in enumerate(modlist):
+        if mod[1] == 'unicodePwd':
+            modlist[idx] = (mod[0], mod[1],
+                map(_format_unicode_password, mod[2]))
```

Comment:
The values are technically a list, I'm not 100% clear on why but that's what the methods in the python ldap module expect. Using map was the most succinct way to accept a list, apply the formatting, and return a list - and as a bonus it applies the formatting to the entire list, if in fact there is a use-case for this of which I'm not aware.